### PR TITLE
Switch to Hugo asset pipeline

### PR DESCRIPTION
**Pull request by lucperkins**
cc @yurishkuro

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,33 +1,22 @@
...
 WRITE_GOOD   := $(NODE_BIN)/write-good
 NODE_VER     := $(shell node -v | cut -c2- | cut -d. -f1)
 GOOD_NODE    := $(shell if [ $(NODE_VER) -ge 4 ]; then echo true; else echo false; fi)
 
 macos-setup: check-node
 	scripts/install-hugo.sh $(HUGO_VERSION) macOS
-	npm install
-	(cd $(THEME_DIR) && npm install)
 
 .PHONY: check-node
```
Comment (yurishkuro) [inline Makefile]: do we still need this?

**Comment by yurishkuro** [comment]
looks like the build is failing
```
12:38:49 AM: ERROR 2018/07/11 04:38:48 error: failed to transform resource: failed to transform "sass/style.sass" (text/x-sass): this feature is not available in your current Hugo version
```

**Comment by lucperkins** [comment]
The version of Hugo with the asset pipeline is not yet available on Netlify, so this will need to wait a bit

**Comment by yurishkuro** [comment]
> The version of Hugo with the asset pipeline is not yet available on Netlify, so this will need to wait a bit

@lucperkins is there an ETA or some other ticket to track that?

**Comment by lucperkins** [comment]
@yurishkuro https://github.com/netlify/binrc/issues/16

**Comment by yurishkuro** [comment]
now blocked by https://github.com/netlify/build-image/issues/183

**Comment by lucperkins** [comment]
@yurishkuro I consulted the lead Hugo dev and got this to work. We should be all set here, as you can see in the preview:

https://deploy-preview-102--jaegertracing.netlify.com/

File: README.md [hunk 2]
```diff
@@ -1,12 +1,5 @@
 # Jaeger website
 
-[doc-img]: https://readthedocs.org/projects/jaeger/badge/?version=latest
-[doc]: http://jaeger.readthedocs.org/en/latest/
-[project]: https://readthedocs.org/projects/jaeger/
-[ci-img]: https://travis-ci.org/jaegertracing/documentation.svg?branch=master
-[ci]: https://travis-ci.org/jaegertracing/documentation
-[jaeger]: https://jaegertracing.io/
-
```
Comment (yurishkuro) [inline README.md]: L7-12 not needed anymore?

File: package.json [hunk 3]
```diff
@@ -1,6 +1,5 @@
 {
   "devDependencies": {
-    "concurrently": "^3.5.1",
```
Comment (yurishkuro) [inline package.json]: should this file go away?

File: resources/_gen/assets/js/js/anchor.min.js_d11fe7b62c27961c87ecd0f2490357b9.content [hunk 4]
```diff
@@ -0,0 +1 @@
...ns.icon),"always"===f&&(d.style.opacity="1"),""===this.options.icon&&(d.style.font="1em/1 anchorjs-icons","left"===this.options.placement&&(d.style.lineHeight="inherit")),"left"===this.options.placement?(d.style.position="absolute",d.style.marginLeft="-1em",d.style.paddingRight="0.5em",n[r].insertBefore(d,n[r].firstChild)):(d.style.paddingLeft="0.375em",n[r].appendChild(d))}for(r=0;r<p.length;r++)n.splice(p[r]-r,1);return this.elements=this.elements.concat(n),this},this.remove=function(A){for(var e,i,n=t(A),o=0;o<n.length;o++)(i=n[o].querySelector(".anchorjs-link"))&&(-1!==(e=this.elements.indexOf(n[o]))&&this.elements.splice(e,1),n[o].removeChild(i));return this},this.removeAll=function(){this.remove(this.elements)},this.urlify=function(A){var t=/[& +$,:;=?@"#{}|^~[`%!'<>\]\.\/\(\)\*\\]/g;return this.options.truncate||e(this.options),A.trim().replace(/\'/gi,"").replace(t,"-").replace(/-{2,}/g,"-").substring(0,this.options.truncate).replace(/^-+|-+$/gm,"").toLowerCase()},this.hasAnchorJSLink=function(A){var e=A.firstChild&&(" "+A.firstChild.className+" ").indexOf(" anchorjs-link ")>-1,t=A.lastChild&&(" "+A.lastChild.className+" ").indexOf(" anchorjs-link ")>-1;return e||t||!1}}});
```
Comment (yurishkuro) [inline resources/_gen/assets/js/js/anchor.min.js_d11fe7b62c27961c87ecd0f2490357b9.content]: where are these `_gen` files coming from? Do we still need a Node.js pipeline to generate them?

Comment (lucperkins) [reply resources/_gen/assets/js/js/anchor.min.js_d11fe7b62c27961c87ecd0f2490357b9.content]: These are generated by Hugo. We'll need to commit this `resources` folder to Git for now, as a workaround for the issues on the Netlify side. Once those are sorted out we can add this to `.gitignore`.

File: README.md (hunk 2 above)
Comment (lucperkins) [reply README.md]: This doesn't actually show up on the [rendered README](https://github.com/jaegertracing/documentation) so I assumed that they're superfluous. If these lines serve some purpose I'd be happy to add them back.

Comment (yurishkuro) [reply README.md]: are you referring to the links above?  They are just shortcuts, don't seem to be used anymore.

File: README.md [hunk 5]
```diff
@@ -1,33 +1,20 @@
 # Jaeger website
 
-[doc-img]: https://readthedocs.org/projects/jaeger/badge/?version=latest
-[doc]: http://jaeger.readthedocs.org/en/latest/
-[project]: https://readthedocs.org/projects/jaeger/
-[ci-img]: https://travis-ci.org/jaegertracing/documentation.svg?branch=master
-[ci]: https://travis-ci.org/jaegertracing/documentation
```
Comment (yurishkuro) [inline README.md]: these two used to be used in the CI badge, I wonder if Netlify provides one.

**Pull request merged**